File: algorithms/[832]翻转图像/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        """
        Iterate
        time O(n^2), space O(1), n 为矩阵长度
        """
        n = len(A)
        for i in range(n):
            for j in range((n + 1) // 2):
                # 遍历一次
                A[i][j], A[i][n - 1 - j] = 1 - A[i][n - 1 - j], 1 - A[i][j]

        return A

File: algorithms/[832]翻转图像/solution.ts
// topics = ["数组"]

/**
 * Iterate
 * time O(n^2), space O(1), n 为矩阵长度
 */
function flipAndInvertImage(A: number[][]): number[][] {
  const n = A.length;
  const m = A[0].length;

  for (let i = 0; i < n; i += 1) {
    A[i].reverse();
    for (let j = 0; j < m; j += 1) {
      A[i][j] = 1 - A[i][j];
      // A[i][j] ^= 1;
    }
  }

  return A;
}
